Clojure 1.3 and Clojure 1.4 compatibility

sivajag opened the pull request:
Changes
- Made it clojure 1.3 and clojure 1.4 compatible
- Upgraded clj-http and clj-oauth2 libs
- Fixed facebook API change in auth.clj
- Using uri.core to make uri

sivajag committed:
updated to clj 1.4beta3

sivajag committed:
ignoring some unneeded files

sivajag committed:
Adding access_token to query params

sivajag committed:
ignoring some unneeded files

sivajag committed:
moved to clojure 1.4

sivajag commented:
Hey 

Did you get a chance to look at my pull request.

-- Siva Jagadeesan

sherbondy commented:
You forgot to include uri in the dependencies.

sivajag commented:
uri is already added as a dependency in clj-oauth project

sherbondy commented:
My mistake. I've been using sivajag's fork of the project recently, and it's been working well for me. I see no reason why it shouldn't be merged.

sivajag commented:
Did you get a chance to look at this pull request?

osbert committed:
Add :grant-type "authorization_code" so that get-access-token works.
May have been broken as a result of an oauth2 library upgrade.

osbert committed:
Use :query-params with string keys.

amitrathore committed:
Merge pull request #1 from osbert/add-grant-type

Add :grant-type "authorization_code" so that get-access-token works.

amitrathore committed:
Merge pull request #2 from osbert/fix-params-key

Use :query-params with string keys.

osbert committed:
Fix [] being returned when no paging data is available.

osbert committed:
Fix mixing of :query-params and :query-string during lazy request
generation based on pagination data. This fixes an infinite looping
problem described below:

When creating a new request from a pagination link, :url is set in the
new request map, with correct parameters like limit, since, etc.

This request is then passed through request middleware, in particular
auth/wrap-facebook-access-token causes :query-params in the request
map to be populated.

This interaction is poor because of how wrap-url and wrap-query-params
interact with each other. wrap-url will deconstruct :url into
:query-string, but wrap-query-params will later overwrite
:query-string with the contents of :query-params.

In our case, this means that the access token will be the only
parameter in the string during each successive lazy request, resulting
in an infinite loop as the offset is lost and we repeatedly request
the first page of results.

The fix implemented is to take the next URL given from Facebook, and
parse it into :query-params instead of relying on :url to propogate
query parameters.

Discovering and triggering this bug is complicated by the fact that it
often will only be triggered if *facebook-auth* is defined when the lazy
evaluation occurs.

For example, this will trigger the bug:

    (with-facebook-auth auth (count (client/get [USER :feed])))

But this will not:

    (count (with-facebook-auth auth (client/get [USER :feed])))

osbert committed:
Add tests for proper lazy request generation and response handling.

osbert committed:
Ignore files created during lein clojars deploy.

amitrathore committed:
Merge pull request #3 from osbert/master

Fix issues with automatic lazy request generation/response handling.
